sdk/dapp-kit/src/reducers/walletReducer.ts

```diff
@@ -11,8 +12,47 @@ export type WalletState = {
 	connectionStatus: 'disconnected' | 'connecting' | 'connected';
 };
 
-export type WalletAction = void;
+type WalletRegisteredAction = {
+	type: 'wallet-registered';
+	payload: {
+		updatedWallets: WalletWithSuiFeatures[];
+	};
+};
+
+type WalletUnregisteredAction = {
+	type: 'wallet-unregistered';
+	payload: {
+		updatedWallets: WalletWithSuiFeatures[];
+		unregisteredWallet: Wallet;
+	};
+};
+
+export type WalletAction = WalletRegisteredAction | WalletUnregisteredAction;
 
-export function walletReducer(state: WalletState): WalletState {
-	return state;
+export function walletReducer(state: WalletState, { type, payload }: WalletAction): WalletState {
+	switch (type) {
+		case 'wallet-registered': {
+			return {
+				...state,
+				wallets: payload.updatedWallets,
+			};
+		}
+		case 'wallet-unregistered': {
+			if (state.currentWallet?.name === payload.unregisteredWallet.name) {
+				return {
```

Comment:
This looks like it's missing setting the updated wallets?

Reply:
Thanks, I'll be adding a test for this specific (albeit super un-realistic) scenario where your current wallet is unregistered and should thus be disconnected once I add the connect mutation hook